### PR TITLE
Use a RichText component for the File block's filename editable content

### LINKED ISSUE #8023: Blocks: File block should use RichText for its contentEditable
Related: https://github.com/WordPress/gutenberg/pull/7622/files#r203397981

The File block currently logs a warning to the browser console:

>Warning: A component is `contentEditable` and contains `children` managed by React. It is now your responsibility to guarantee that none of those nodes are unexpectedly modified or duplicated. This is probably not intentional.

In having need for ContentEditable, we should be promoting use of the RichText component, which includes abstractions necessary to deal with its inherent complexities. 

https://github.com/WordPress/gutenberg/tree/master/editor/components/rich-text

Using RichText should also resolve the warning.

### PR DESCRIPTION
**Pull request by talldan**
## Description
Closes #8023 

File block was using a `contenteditable` anchor element for its editor link text. This replaces the anchor element's content with a `RichText`

## How has this been tested?
Manual testing

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by talldan** [comment]
@noisysocks - I wonder if it's worth getting rid of the anchor element entirely and not worrying about the copy/paste behaviour. It's a bit weird having to put a div inside of an anchor tag. Also, preventing the anchor link from opening is weird.

I've also noticed something weird going on with file blocks - I sometimes get a message 'This block appears to have been modified externally.' when it hasn't. Eeek.

File: core-blocks/file/editable-link.js [hunk 1]
```diff
@@ -39,40 +29,31 @@ export default class FileBlockEditableLink extends Component {
...
+					onClick={ ( event ) => event.preventDefault() }
 					onCopy={ this.copyLinkToClipboard }
 					onCut={ this.copyLinkToClipboard }
 					onPaste={ this.forcePlainTextPaste }
-					contentEditable
 				>
-					{ text }
+					<RichText
+						wrapperClassName={ `${ className }__textlink-richtext` }
+						tagName="div" // must be block-level or else cursor disappears
```
Comment (noisysocks) [inline core-blocks/file/editable-link.js]: Why does the cursor disappear?

**Comment by noisysocks** [comment]
Yeah, I think you're right. It feels really unintuitive having the cursor be a hand instead of a vertical bar.

File: core-blocks/file/index.js [hunk 2]
```diff
@@ -217,7 +217,7 @@ export const settings = {
 					<a
 						href={ href }
 						className="wp-block-file__button"
-						download={ fileName }>
+						download={ fileName || '' }>
```
Comment (noisysocks) [inline core-blocks/file/index.js]: Why is this needed?

**Review by noisysocks — APPROVED** [review]
Love it! 😍

File: core-blocks/file/index.js (hunk 2 above)
Comment (talldan) [reply core-blocks/file/index.js]: I had an issue before whereby if the fileName is undefined, the download attribute is not added to the resulting anchor, which changes the behaviour of the button from downloading a file to opening it in a new tab. It's probably not needed now. Not sure if there could ever be an edge-case where a fileName could be undefined now that the string format is being used for the rich text field.

Comment (noisysocks) [reply core-blocks/file/index.js]: Fair enough. I wonder if it should be `fileName || true` so that if there is no file name, the attribute is added but not set to anything? e.g. `<a href="..." download>`
